bookstore-service/src/main/java/org/people/weijuly/bookstore/graphql/impl/mutation/LendBooksMutationResolverImpl.java

```diff
@@ -72,13 +78,27 @@ private void validate(CartInModel cartIn) {
                         String.format("Customer with id: %s cannot be found", cartIn.getCustomerId()),
                         HttpStatus.BAD_REQUEST
                 ));
-        for (String bookId : cartIn.getBooks()) {
+        Set<String> booksSet = new HashSet<>(booksInCart);
+        if (booksSet.size() < booksInCart.size()) {
```

Comment:
Instead of creating new variable, we can check like this `if (cartIn.getBooks().size() != new HashSet<>(cartIn.getBooks()).size())`